src/libs/antares/benchmarking/include/antares/benchmarking/DurationCollector.h

```diff
@@ -30,34 +32,42 @@
 namespace Benchmarking
 {
 
-class IDurationCollector
+class DurationCollector
 {
 public:
-    virtual ~IDurationCollector() = default;
-    virtual void addDuration(const std::string& name, int64_t duration) = 0;
-};
+    void toFileContent(FileContent& file_content);
+    void addDuration(const std::string& name, long duration);
 
-class NullDurationCollector : public IDurationCollector
-{
-public:
-    NullDurationCollector() = default;
-    virtual ~NullDurationCollector() = default;
-    void addDuration(const std::string& /* name */, int64_t /* duration */) override
-    { /* Do nothing */
+    using clock = std::chrono::steady_clock;
+
+    struct OperationTimer
+    {
+        OperationTimer(DurationCollector& collector, const std::string& key):
+            collector(collector), key(key) {}
+
+        DurationCollector& collector;
+        const std::string key;
+    };
+
+    OperationTimer operator()(const std::string& key)
+    {
+        return OperationTimer(*this, key);
     }
-};
 
-class DurationCollector : public IDurationCollector
-{
-public:
-    DurationCollector() = default;
-    virtual ~DurationCollector() = default;
+    friend void operator<<(const OperationTimer& op, const std::function<void(void)>& f)
+    {
+        static std::mutex mutex;
```

Comment:
`static` is very bad. `static` means this object will be shared across threads, across all objects. If you really need it, define the mutex next to the `std::map`.